[Term Entry] Python Pillow - Image: .putalpha() (#6384)
Codecademy/docs: content/pillow/concepts/image/terms/putalpha/putalpha.md

---
Title: '.putalpha()'
Description: 'Modifies the alpha channel (transparency) of an image in the Pillow library.'
Subjects:

>   - 'Computer Science'

Reviewer: Add more subjects here - 

```suggestion
  - 'Computer Science'
  - 'Data Science'
  - 'Data Visualization'
```

Tags:
  - 'Python'
  - 'Pillow'

>   - 'Image Processing'

Reviewer: 'Image Processing' is not a valid tag, add something like these - 

```suggestion
  - 'Computer Vision'
  - 'Images'
  - 'Values'
```

CatalogContent:
  - 'learn-python'
  - 'paths/computer-science'
---

> ## Description

Reviewer: We don't add heading for description.

```suggestion
```

> The `.putalpha()` method in the **Pillow** library modifies the alpha channel of an image, controlling its transparency. This method allows setting a uniform transparency level or using another image as an alpha mask.

Reviewer: The first mention of the method should be emphasized in bold. 

```suggestion
The **`.putalpha()`** method in the Pillow library modifies the alpha channel of an image, allowing control over its transparency. This method can either apply a uniform transparency level to the entire image or use another image as an alpha mask.

```

## Syntax

> ```python

Reviewer: Syntax should be inside `pseudo` codeblock - 

```suggestion
```pseudo
```

image.putalpha(alpha)
```

### Parameters:

> - `alpha` *(int or Image)*: The alpha value to be applied.
>   - If an integer (0-255), it sets a uniform transparency level for the entire image (0 is fully transparent, 255 is fully opaque).
>   - If an `Image` object, it applies the alpha values from that image as a transparency mask.

Reviewer: This can be better written as -

```suggestion
- **`alpha`** (`int` or `Image`) – Defines the transparency level of the image.
  - **`int`** (`0-255`): Applies uniform transparency across the entire image (`0` = fully transparent, `255` = fully opaque).
  - **`Image` object**: Uses the pixel values from the provided image as an alpha mask, determining varying levels of transparency.
```

> ## Example

Reviewer: The two examples should be structured as per the [style guide](https://github.com/Codecademy/docs/blob/main/documentation/style-guide.md). It can be like - 

## Example 1
## Example 2

or 

## Examples

### Applying a Uniform Transparency Level
### Using an Alpha Mask

Please update accordingly.

> ```python

Reviewer: ```suggestion
```py
```

from PIL import Image

# Open an existing image
image = Image.open("example.png").convert("RGBA")

# Modify the alpha channel (50% transparency)
image.putalpha(128)

# Save the new image
image.save("example_transparent.png")
```

### Example Using an Alpha Mask

```python
# Create an RGBA image and an alpha mask
alpha_mask = Image.open("mask.png").convert("L")
image.putalpha(alpha_mask)
image.save("example_masked.png")

> ```

Reviewer: For a better understanding, you should include the example input and output images in the entry for both examples.

## Notes
- The `.putalpha()` method only works on images in **RGBA** or **LA** mode.
- When using an alpha mask, the mask must be in **L** (grayscale) or **1** (black-and-white) mode.

> ## Related Methods
> - `.convert("RGBA")` – Ensures the image has an alpha channel before using `.putalpha()`.
> - `.split()` – Separates an image into its component bands, including the alpha channel.
> 
> ## Resources
> - [Pillow Documentation: Image.putalpha()](https://pillow.readthedocs.io/en/stable/reference/Image.html#PIL.Image.Image.putalpha)

Reviewer: We can remove these sections as they will be covered in other doc entries -

```suggestion
```

## Syntax

```psuedo
image.putalpha(alpha)
```

### Parameters:
  - **`alpha`** (`int` or `Image`): Defines the transparency level of the image.
    - **`int`** (`0-255`): Applies uniform transparency across the entire image (`0` = fully transparent, `255` = fully opaque).
    - **`Image` object**: Uses the pixel values from the provided image as an alpha mask, determining varying levels of transparency.

## Example 1: Applying Uniform Transparency

```py
from PIL import Image

# Open an existing image
image = Image.open("example.png").convert("RGBA")

# Modify the alpha channel (50% transparency)
image.putalpha(128)

# Save the new image
image.save("example_transparent.png")
```

## Example 2: Using an Alpha Mask

``` py 
from PIL import Image 

# Open an existing image  
image = Image.open("example.png").convert("RGBA")  

# Create an RGBA image and an alpha mask  
alpha_mask = Image.open("mask.png").convert("L")  
image.putalpha(alpha_mask)  

# Save the new image  

> image.save("example_masked.png") 

Reviewer: Hey @subigya-prasai, are you working on this?

> image.save("example_masked.png") 

Author: Yes, sorry i was not able to work as i was out for whole 2 days.
